Table component: adding click event on expandable table rows

MarionHauffNAV opened the pull request:
### Description

Table component: adding click event on expandable table rows

Ref Slack tråd https://nav-it.slack.com/archives/C7NE7A8UF/p1685454962205769)


### Change summary

- added click event on table row
- fixed styling on table row when shadeOnHover prop is true and expandable is disabled (hover seemed confusing as the table has no interaction)

mlp73 committed:
Table component: adding click event on expandable table rows

mlp73 committed:
Merge branch 'main' into mh/expandable-table-click-area

it-vegard commented on @navikt/core/css/table.css:
```diff
@@ -154,6 +154,10 @@
   transition: border-bottom-color 190ms cubic-bezier(0.6, 0.04, 0.98, 0.335);
 }
 
+.navds-table__expandable-row:not(.navds-table__expandable-row--open):not(.navds-table__expandable-row--expansion-disabled):hover {
```
Foreslår å fjerne unntaket for når den er åpen, slik at pointer-cursor vises også når brukeren skal lukke raden.

```suggestion
.navds-table__expandable-row:not(.navds-table__expandable-row--expansion-disabled):hover {
```

Commit:
Update @navikt/core/css/table.css

Fjerner unntaket for når raden er åpen, slik at pointer-cursor vises også når brukeren skal lukke raden.

Co-authored-by: Vegard Haugstvedt <it.vegard@gmail.com>

KenAJoh committed:
:test_tube: Interaktive elementer test

KenAJoh commented on @navikt/core/react/src/table/stories/table-expandable.stories.tsx:
```diff
@@ -227,3 +230,72 @@ export const ExpandableOpen = () => {
+      </Table>
+    );
+  },
+  play: async ({ canvasElement, step }) => {
```
Laget en interaktiv demo (kan ses i storybook) for et av problemene som har stoppet oss fra denne oppdateringen tidligere:

I tilfeller der man har andre interaktive elementer som eks lenker, checkboxer eller knapper ønsker vi at man ikke skal åpne raden ved klikk på dem. Dette kan kanskje også videreføres til at raden ikke skal åpnes hvis man ønsker å eks kopiere noe tekst, men ikke sikker på den.

Arbeidet stoppet litt opp med å finne en god løsning på dette, men kan hende alt man trenger er å sjekke `document.activeElement` 🤞 Vært helt mega å finne en god løsning på dette da jeg vet mange ønsker seg denne featuren 🙌

mlp73 committed:
Legger til stop propagation på interkative elementer innen en expandable row

mlp73 committed:
Table component: prevent expand when clicking other elements than the row itself or the expandable button

it-vegard commented on @navikt/core/react/src/table/ExpandableRow.tsx:
```diff
@@ -65,17 +65,40 @@ export const ExpandableRow: ExpandableRowType = forwardRef(
+
+    const onRowClick = (e) => {
+      if (
+        e.target.nodeName === "TD" ||
```
Selv om vi ikke har noen kodeeksempler på det (og stylingen for det virker litt "off"), så skulle jeg gjerne ønsket at vi sjekker for "TH" i tillegg til "TD", i tilfelle noen bruker radoverskrifter. Mest for å forhindret at vi glemmer dette om man tar i bruk radoverskrifter i fremtiden.

it-vegard commented on @navikt/core/react/src/table/ExpandableRow.tsx:
```diff
@@ -65,17 +65,40 @@ export const ExpandableRow: ExpandableRowType = forwardRef(
+      if (
+        e.target.nodeName === "TD" ||
+        e.target === expandableButtonRef.current ||
+        e.target === expandableIconRef.current
```
Trenger vi disse to sjekkene, siden knappen bruker expansionHandler direkte? 
onRowClick trenger ikke håndtere knappetrykk slik jeg forstår koden.
Da kan vi fjerne de to ref-ene også.

mlp73 committed:
opprydning i logikken for klikk på expandable row

mlp73 committed:
fjerner utkommentert kode

KenAJoh commented on @navikt/core/react/src/table/ExpandableRow.tsx:
```diff
@@ -65,17 +65,33 @@ export const ExpandableRow: ExpandableRowType = forwardRef(
+            "navds-table__expandable-row--expansion-disabled":
+              expansionDisabled,
           })}
+          onClick={!expansionDisabled ? onRowClick : undefined}
```
```suggestion
          onClick={(e) => {
          !expansionDisabled && onRowClick(e);
          rest?.onClick?.(e);
          }
```

Commit:
Update @navikt/core/react/src/table/ExpandableRow.tsx

Co-authored-by: Ken <26967723+KenAJoh@users.noreply.github.com>

mlp73 committed:
syntax fix

mlp73 committed:
Table component: legger til prop expandOnRowClick

KenAJoh commented on .changeset/heavy-pandas-marry.md:
```diff
@@ -0,0 +1,6 @@
+"@navikt/ds-react": minor
+---
+
+Adding click event on table row
```
```suggestion
Table: ExpandableRow kan nå åpnes med 'expandOnRowClick'-prop 
```

KenAJoh committed:
Update .changeset/heavy-pandas-marry.md

KenAJoh reviewed (approved):
🚢 🇮🇹 Ser bra ut!

mlp73 committed:
Merge branch 'main' into mh/expandable-table-click-area
